enhancements/cluster-scope-secret-volumes/shared-resources-via-openshift-builds-operator.md

```diff
@@ -0,0 +1,291 @@
+
+- Provides an extension to OpenShift Builds, which is based on the upstream [Shipwright](https://shipwright.io) project.
+- Allows release outside of OpenShift's cadence.
+- Potential to fully support customers on older versions of OCP.
+- Pathway to adoption on hosted control plane clusters.
+
+
+### User Stories
+
+
+- As a developer installing RHEL RPMs in my container image, I want to use the Shared Resource CSI
+  driver to mount my cluster's entitlement into my build.
+- As a cluster admin of a build/CI cluster, I want to manage the Shared Resource CSI driver via an
+  operator so that its installation, custom resources, and lifecycle are managed by OLM.
+- As an SRE operating hosted control plane clusters, I want the Shared Resource CSI driver to be
+  managed by OLM so that its admission webhooks are not run on the hosted control plane.
+- As a Red Hat software engineer, I want to release the Shared Resource CSI Driver via OLM so that
+  I can fully support the driver on more versions of OCP.
+
+
+### Goals
+
+- Deploy the Shared Resource CSI Driver via OLM
+- Deprecate and remove the Shared Resource CSI Driver as a tech preview feature in OpenShift.
+
+
+### Non-Goals
+
+- Automatically mount RHEL entitlements into builds.
+- Deploy OLM operator webhooks on hosted control planes.
+- Describe the low level implementation of the OpenShift Builds Operator.
+- Management of the Build capability in OCP.
+
+
+## Proposal
+
+The `CSIDriverSharedResource` feature will be officially deprecated in OCP 4.15, and removed fully
+in OCP 4.16. A blog post will announce this deprecation and its replacement with the OpenShift Builds operator.
+
+The OpenShift Builds operator will be enhanced to support the lifecycle and management of the
+Shared Resource CSI Driver and its associated components. This will be in addition to the
+lifecycle management of Shipwright, whose logic is implemented in the upstream
+[Shipwright Operator](https://github.com/shipwright-io/operator).
+
+
+### Workflow Description
+
+1. Cluster admin installs the OpenShift Builds operator via OLM, using one of the following:
+   1. OperatorHub web console in OpenShift
+   2. Creating an appropriate OLM `Subscription` object via `oc apply`, OpenShift GitOps, and so
+      forth.
+2. Once the operator is deployed, it creates an `OpenShiftBuild` custom resource instance on the
+   cluster:
+
+   ```yaml
+   apiVersion: operator.build.openshift.io/v1alpha1
+   kind: OpenShiftBuild
+   metadata:
+     name: cluster
+   spec:
+     shipwright:
+       build:
+         enabled: false
+     sharedResources:
+       enabled: false
+   ```
+
+3. The OpenShift Builds Operator reconciles the `OpenShiftBuild` instance (singleton), and reports
+   an appropriate status. If `spec.sharedResources.enabled` is `true`, the operator deploys the
+   Shared Resource CSI driver, custom resource definitions, and webhook.
```

Comment:
what happens when this is switched back to "false" after having been set to true?

Reply:
My immediate thought to answering @bparees 's question is that "shared resources is uinstalled"

But then how much do we worry about, or can we even properly, "fail gracefully" for BuildConfig's that remain that point to the shared resource CSI driver for one of its volumes?  Does feel like perhaps the EP has to lay out the plan for that.

Does it somehow update any BuildConfigs using shared resources CSI volumes?

Reply:
@jsafrane may have to weigh in on the "right" way to remove a CSI driver.